powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/ClientEciesEncryptor.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2023 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.security.powerauth.crypto.lib.encryptor.ecies;
+
+import io.getlime.security.powerauth.crypto.lib.encryptor.ClientEncryptor;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ecies.exception.EciesException;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ecies.model.EciesCryptogram;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ecies.model.EciesParameters;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ecies.model.EciesPayload;
+import io.getlime.security.powerauth.crypto.lib.encryptor.exception.EncryptorException;
+import io.getlime.security.powerauth.crypto.lib.encryptor.model.*;
+import io.getlime.security.powerauth.crypto.lib.encryptor.model.v3.ClientEncryptorSecrets;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.model.exception.CryptoProviderException;
+import io.getlime.security.powerauth.crypto.lib.util.EciesUtils;
+import lombok.Getter;
+
+import java.util.Base64;
+
+/**
+ * Class implements ECIES encryption for PowerAuth clients.
+ * <p>PowerAuth protocol versions:
+ * <ul>
+ *     <li>3.0</li>
+ *     <li>3.1</li>
+ *     <li>3.2</li>
+ * </ul>
+ */
+public class ClientEciesEncryptor implements ClientEncryptor {
+
+    private static final KeyGenerator keyGenerator = new KeyGenerator();
+
+    private final EncryptorId encryptorId;
+    private final EncryptorParameters encryptorParameters;
+    private final EciesRequestResponseValidator validator;
+    private final byte[] associatedData;        // non-null for V3.2+
+
+    // Variables altered after configureKeys() call.
+    private ClientEncryptorSecrets encryptorSecrets;
+
+    /**
+     * Get SharedInfo2 base bytes. The function should be used only for the testing purposes.
+     */
+    @Getter
```

Comment:
Not sure if this light usage of Lombok is necessary.

Reply:
Good catch. At some point I needed the values in tests, but then I changed how we test the encryption in general.